portafly/src/components/shared/data-list/reducers/DataReducer.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  ActionHandlers,
+  Action,
+  createReducer
+} from 'utils'
+import { IDeveloperAccount, IApplication } from 'types'
+
+// Union Type of the different Data Types a DataListContext will hold
+export type DataState = IDeveloperAccount[] | IApplication[]
+
+// Action Handlers
+const SET_DATA = 'SET_DATA'
+
+type SetDataAction = Action<DataState>
+
+const filtersActionHandlers: ActionHandlers<DataState, DataState> = {
+  [SET_DATA]: (state, action: SetDataAction) => action.payload
+}
+
+// Reducer
+const dataReducer = createReducer(filtersActionHandlers)
+
+// Hook
+interface IUseData {
+  state: Record<'data', DataState>
+  dispatch: React.Dispatch<Action<DataState>>
+}
```

Comment:
Instead of a union type for `DataSet`, I think we should use generics:
```suggestion
interface IUseData<T> {
  state: Record<'data', <T>>
  dispatch: React.Dispatch<Action<T>>
}
```

Reply:
Good thing of generics is less code and we don't need to define what to expect... but the bad thing is it could accept everything that is passed to the interface. If we think we shouldn't constrain what data is ok to be stored, then it's ok to go for generics.

Reply:
I think we don't have any constraints 🤔 What defines a DataList? A list of data, where data is a collection of anything, really.

Reply:
In our data list, it has some specifics. There's no usability without id and at least a matching row - column